src/query_plan/operation.rs

```diff
@@ -869,6 +891,34 @@ pub(crate) mod normalized_field_selection {
             &mut self.data.sibling_typename
         }
 
+        pub(crate) fn type_if_added_to(
+            &self,
+            parent_type: &CompositeTypeDefinitionPosition,
+            schema: &ValidFederationSchema,
+        ) -> Option<CompositeTypeDefinitionPosition> {
+            let data = self.data();
+            if &data.field_position == parent_type {
```

Comment:
this should be returning something like
```rs
if data.field_position.parent() == *parent_type && data.schema == *schema {
  let field = data.field_position.get(schema.schema())?;
  return Ok(Some(field.ty.clone()));
}
```

Reply:
`field.ty` is an AST `Type`, not a composite type. Knowing where this method is called, we could try looking up the `ExtenedType` and then converting that into a composite type, and flatten everything using `Result::ok` (if this method would return anything other than `Some`/`Ok(Some())`, the calling method will return `false`, so dropping the `Err` doesn't not cause any info loss). This would look (roughly) like this:
```rust
if &data.field_position.parent() == parent_type && &data.schema == schema {
  let base_ty_name = data.field_position.get(schema.schema()).ok()?.ty.inner_named_type();
   return schema.get_type(base_ty_name.clone()).and_then(CompositeTypeDefinitionPosition::try_from).ok();
}
```